### PR TITLE
Aggregation not work correctly

### LINKED ISSUE #2151: Aggregation not send correct values
**Informations**
- **Node.js version**: v9.11.1
- **npm version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.14.3 
- **Database**: mongo 4.01
- **Operating system**: macOS 10.14

**What is the current behavior?**
using the ${model}Connection query I do not receive the correct values

**Steps to reproduce the problem**
I'm trying to get the number of documents from a query. I have a Contact model, so I have the following query:
contacts_aggregation

(I know that only 2 documents fulfill this filtering.) The count is correct but in the values ​​those fields that I used for the filtering are not solved correctly.

**What is the expected behavior?**
Get the correct count and the correct values.
At the moment I had to do this:
resultadoesperado

**Suggested solutions**
no suggestion

### PR DESCRIPTION
**Pull request by jrperdomoz**
My PR is a:
🐛 Bug fix #2151

Main update on the:
Plugin: graphql

Values ​​and counting are resolved correctly.  I solve the errors detailed in the number #2151

### CODE REVIEW

**Comment by jrperdomoz** [comment]
I think that this "buildOr" function should be in the "strapi-utils" package to be reused, because I have seen that this code fragment is repeated in the search functions of the api services. What do you think @Aurelsicoko ?

**Comment by jrperdomoz** [comment]
I think that the "buildOr" function should go in the "strapi-utils" package, since that code fragment is used in the search function of the different apis that are created and also here it is also needed.
What do you think @Aurelsicoko ?

**Comment by Aurelsicoko** [comment]
@jrperdomoz I looked at your changes very quickly, it seems good to me. Are you sure the `buildOr` method is generic? I mean it works with Mongoose, and Bookshelf, and any other connector in the future.

**Comment by jrperdomoz** [comment]
@Aurelsicoko The "buildOr" function only creates an operation $or of mongo, so it would only work for mongoose. Then what would be the ideal place for this function?

**Comment by lauriejim** [comment]
@jrperdomoz I think the question is _Did we need to move this function in utils ?_

**Comment by jrperdomoz** [comment]
@lauriejim I don't know which is the ideal place to move the function, but if I think that for the sake of not repeating code needlessly it should be in a place of easy access. This same code snippet I've looked at it in the APIs.

**Comment by jrperdomoz** [comment]
@lauriejim @Aurelsicoko  I move the buildOr function again to Aggregator.js

**Comment by Aurelsicoko** [comment]
The right place should be to move it in the Mongoose hook utils (see https://github.com/strapi/strapi/tree/master/packages/strapi-hook-mongoose/lib/utils). What do you think about that?

**Comment by Aurelsicoko** [comment]
We should be able to call the function through `strapi.hook.mongoose.utils.buildOr()`. However, the utils are not exposed in the `strapi.hook.mongoose` object. It means we need to edit the file https://github.com/strapi/strapi/blob/master/packages/strapi/lib/core/hooks.js like this

```js
'use strict';

// Dependencies.
const path = require('path');
const glob = require('glob');
const { parallel } = require('async');
const { endsWith, get, set } = require('lodash');

module.exports = function() {
  this.hook = {};

  return Promise.all([
    new Promise((resolve, reject) => {
      const cwd = this.config.appPath;

      // Load configurations.
      glob('./node_modules/*(strapi-hook-*)/!(node_modules)/**/*.*(js|json)', {
        cwd
      }, (err, files) => {
        if (err) {
          return reject(err);
        }

        mountHooks.call(this, files, cwd, 'node_modules')(resolve, reject);
      });
    }),
    new Promise((resolve, reject) => {
      const cwd = path.resolve(this.config.appPath, 'hooks');

      // Load configurations.
      glob('./**/*.*(js|json)', {
        cwd
      }, (err, files) => {
        if (err) {
          return reject(err);
        }

        mountHooks.call(this, files, cwd)(resolve, reject);
      });
    }),
    new Promise((resolve, reject) => {
      const cwd = path.resolve(this.config.appPath, 'plugins');

      // Load configurations.
      glob('./*/hooks/**/*.*(js|json)', {
        cwd
      }, (err, files) => {
        if (err) {
          return reject(err);
        }

        mountHooks.call(this, files, cwd, 'plugins')(resolve, reject);
      });
    })
  ]);
};

const mountHooks = function (files, cwd, source) {
  return (resolve, reject) =>
    parallel(
      files.map(p => cb => {
        const folders = p.replace(/^.\/node_modules\/strapi-hook-/, './')
          .split('/');
        const name = source === 'plugins' ? folders[folders.length - 2] : folders[1];

        this.hook[name] = this.hook[name] || {
          loaded: false
        };

        let dependencies = [];
        if (source === 'node_modules') {
          try {
            dependencies = get(require(path.resolve(this.config.appPath, 'node_modules', `strapi-hook-${name}`, 'package.json')), 'strapi.dependencies', []);
          } catch(err) {
            // Silent
          }
        }

        if (endsWith(p, 'index.js') && !this.hook[name].load) {
          // Lazy loading.
          Object.defineProperty(this.hook[name], 'load', {
            configurable: false,
            enumerable: true,
            get: () => require(path.resolve(cwd, p)),
            dependencies
          });

          this.hook[name].dependencies = dependencies;
        } else {
          const splitPath = p.split('/');
          const objectPath = splitPath.slice(splitPath.indexOf('lib') + 1).map(name => name.replace('.js', ''));

          set(this.hook, [name].concat(objectPath), require(path.resolve(cwd, p)));
        }

        cb();
      }),
      (err) => {
        if (err) {
          return reject(err);
        }

        resolve();
      }
    );
};
```

Then, you'll be able to retrive the utils via `strapi.hook.mongoose.utils.buildOr()`.

**Comment by jrperdomoz** [comment]
@Aurelsicoko  I think everything is correct. 
buildor

But after running the `npm run setup` command and running my project I have this bug: #2251

File: packages/strapi-plugin-graphql/services/Aggregator.js [hunk 1]
```diff
@@ -75,22 +77,6 @@ module.exports = {
     }, {});
   },
 
-  /**
-   * Returns a list of fields that have type included in fieldTypes.
-   */
-  getFieldsByTypes: (fields, typeCheck, returnType) => {
```
Comment (lauriejim) [inline packages/strapi-plugin-graphql/services/Aggregator.js]: Why did you delete this function ?
It still call many time `this. getFieldsByTypes`

**Comment by lauriejim** [comment]
This update break the plugin. 
As expected in my comment, I get `(node:7111) UnhandledPromiseRejectionWarning: TypeError: this.getFieldsByTypes is not a function`

File: packages/strapi-plugin-graphql/services/Aggregator.js (hunk 1 above)
Comment (jrperdomoz) [reply packages/strapi-plugin-graphql/services/Aggregator.js]: because it was duplicated, there was another exactly the same

Comment (jrperdomoz) [reply packages/strapi-plugin-graphql/services/Aggregator.js]: I'm sorry it was a mistake, I've already corrected it.

**Comment by lauriejim** [comment]
@jrperdomoz Can you told me why you use `/* eslint-disable no-unused-vars */` in many files ?

**Comment by jrperdomoz** [comment]
> @jrperdomoz Can you told me why you use `/* eslint-disable no-unused-vars */` in many files ?

@lauriejim Because a few days ago the code did not pass all the checks, then I saw several alerts before doing the commit. similar to this:
captura de pantalla 2018-11-13 a las 9 31 33

If it's a nuisance, I can eliminate them

**Comment by lauriejim** [comment]
Yes please can you remove them and make sure pre-commit is OK.
Appreciate your work. I will test your PR when it's done.

> Also check empty space https://github.com/strapi/strapi/pull/2157/files#diff-9f9e717ab72a3b5e13d21d3ceff70428R94

**Comment by lauriejim** [comment]
Will be fixed by this more massive PR https://github.com/strapi/strapi/pull/2452
In any case thank you for your time. It's appreciate.
